providers/implementations/keymgmt/ecx_kmgmt.c

```diff
@@ -612,11 +733,31 @@ static void *ecx_gen(struct ecx_gen_ctx *gctx)
         if (!ossl_ed25519_public_from_private(gctx->libctx, key->pubkey, privkey,
                                               gctx->propq))
             goto err;
+#ifdef FIPS_MODULE
+        if (!fips_eddsa_pct(gctx->libctx,
+                            ECX_KEY_TYPE_ED25519,
+                            key->pubkey,
+                            privkey,
+                            key->keylen)) {
+            /* rsa pct aborts on fail, do the same. */
+            abort();
+        }
+#endif
         break;
     case ECX_KEY_TYPE_ED448:
         if (!ossl_ed448_public_from_private(gctx->libctx, key->pubkey, privkey,
                                             gctx->propq))
             goto err;
+#ifdef FIPS_MODULE
+        if (!fips_eddsa_pct(gctx->libctx,
+                            ECX_KEY_TYPE_ED448,
+                            key->pubkey,
+                            privkey,
+                            key->keylen)) {
+            /* rsa pct aborts on fail, do the same. */
```

Comment:
comment states RSA should be EdDSA

Reply:
What I meant in the comment is: "In generate key, the RSA pct test calls abort() when it fails" - so I cause the eddsa PCT test to do the same. Sorry if this wasn't clear (I thought it was).